studio/src/editor-panel.js

```diff
@@ -208,15 +227,79 @@ export default class EditorPanel extends LitElement {
         if (this.needsMask(store.get(id))) {
             this.maskOtherFragments(id);
         }
-        // Load fragment versions when opening a fragment
         this.loadFragmentVersions();
+        await this.loadParentFragmentContext(id);
+    }
+
+    async loadParentFragmentContext(fragmentId) {
+        this.editorContextStore.reset();
+        this.parentFragment = null;
+        this.parentFragmentLoading = false;
+
+        try {
+            await this.editorContextStore.loadFragmentContext(fragmentId);
+            if (this.editorContextStore.hasParent()) {
+                await this.fetchParentFragment();
+            }
+        } catch (error) {
+            console.error('Failed to load fragment context:', error);
+        }
+    }
+
+    async fetchParentFragment() {
+        const parentId = this.editorContextStore.getParentId();
+        if (!parentId || parentId === this.fragment?.id) {
+            this.parentFragment = null;
+            this.parentFragmentLoading = false;
+            return;
+        }
+
+        this.parentFragmentLoading = true;
+        try {
+            const parentData = await this.repository.aem.sites.cf.fragments.getById(parentId);
+            this.parentFragment = new Fragment(parentData);
+        } catch (error) {
+            console.error('Failed to fetch parent fragment:', error);
+            showToast(`Failed to load parent fragment: ${error.message}`, 'negative');
+            this.parentFragment = null;
+        } finally {
+            this.parentFragmentLoading = false;
+        }
+    }
+
+    async navigateToParentFragment() {
+        if (!this.parentFragment) return;
+        await this.closeEditor();
+        await router.navigateToFragmentEditor(this.parentFragment.id);
     }
 
     handleKeyDown(event) {
         if (event.code === 'Escape') this.closeEditor();
         if (!event.ctrlKey) return;
+
         if (event.code === 'ArrowLeft' && event.shiftKey) this.updatePosition('left');
         if (event.code === 'ArrowRight' && event.shiftKey) this.updatePosition('right');
+
+        if (event.code === 'KeyS') {
+            event.preventDefault();
+            if (Store.editor.hasChanges) this.saveFragment();
+        }
+        if (event.code === 'KeyU') {
+            event.preventDefault();
+            this.publishFragment();
+        }
+        if (event.code === 'KeyL') {
+            event.preventDefault();
+            this.showClone();
+        }
+        if (event.code === 'KeyK') {
+            event.preventDefault();
+            this.copyToUse();
+        }
+        if (event.code === 'Backspace') {
+            event.preventDefault();
+            this.deleteFragment();
+        }
```

Comment:
:-) this is awesome (but could have been done in another PR to speed up this review ;) )